tests/conftest.py

```diff
@@ -154,6 +154,18 @@ def mock_connect(monkeypatch):
     yield connect
 
 
+@pytest.fixture(scope="session", autouse=True)
+def hide_window(request):
+    old_value = os.environ.get("QT_QPA_PLATFORM", None)
+    if not request.config.getoption("--show-gui"):
+        os.environ["QT_QPA_PLATFORM"] = "offscreen"
+    yield
+    if old_value:
```

Comment:
If `old_value` is `""`, then it's still falsey.

I think you should rewrite the whole function to:
```py
def hide_window(request):
    if request.config.getoption("--show-gui"):
        return

    old_value = os.environ.get("QT_QPA_PLATFORM")
    os.environ["QT_QPA_PLATFORM"] = "offscreen"
    yield
    if old_value is None:
        del os.environ["QT_QPA_PLATFORM"]
    else:
        os.environ["QT_QPA_PLATFORM"] = old_value
```

Reply:
Agree, that reads better